### PR TITLE
Annotation option

### PR DESCRIPTION
**Pull request by remijouannet**
Hello

A quick and easy patch to add an Annotation option for iso and cmd builder to edit the "note" part of the VM

### CODE REVIEW

File: clone/step_clone.go [hunk 1]
```diff
@@ -26,6 +27,10 @@ func (c *CloneConfig) Prepare() []error {
 		errs = append(errs, fmt.Errorf("'linked_clone' and 'disk_size' cannot be used together"))
 	}
 
+	if c.Annotation == "" {
+		c.Annotation = "generate by jetbrains-infra/packer-builder-vsphere"
```
Comment (VladRassokhin) [inline clone/step_clone.go]: `s/generate/generated/` maybe?

Comment (mkuzmin) [reply clone/step_clone.go]: I'd prefer empty. Taking a snapshot we have to set some name. Here it's not required.

File: README.md [hunk 2]
```diff
@@ -48,6 +48,7 @@ See complete Ubuntu, Windows, and macOS templates in the [examples folder](https
 ### VM Location
 
 * `vm_name`(string) - Name of the new VM to create.
+* `annotation`(string) - Add some note.
```
Comment (mkuzmin) [inline README.md]: in UI it's named "Notes". We probably should follow end users, not internal APIs.

File: iso/step_create.go [hunk 3]
```diff
@@ -67,6 +73,7 @@ func (s *StepCreateVM) Run(_ context.Context, state multistep.StateBag) multiste
 		USBController:       s.Config.USBController,
 		Version:             s.Config.Version,
 		Firmware:            s.Config.Firmware,
+		Annotation:          s.Config.Annotation,
```
Comment (mkuzmin) [inline iso/step_create.go]: interesting, we already have it in the driver.

Comment (remijouannet) [reply iso/step_create.go]: yeah i had the same reaction
https://github.com/jetbrains-infra/packer-builder-vsphere/commit/ba06b00e70d1c705835ea4894ec101e19f18b11b
I think you just forgot to copy/paste the line for annotation in this commit

File: driver/vm.go [hunk 4]
```diff
@@ -219,6 +220,10 @@ func (template *VirtualMachine) Clone(ctx context.Context, config *CloneConfig)
 		cloneSpec.Snapshot = tpl.Snapshot.CurrentSnapshot
 	}
 
+	if config.Annotation != "" {
+		cloneSpec.Config.Annotation = config.Annotation
```
Comment (mkuzmin) [inline driver/vm.go]: `cloneSpec.Config` is undefined:
```
GOROOT=/usr/local/lib/go-1.10.4 #gosetup
GOPATH=/Users/mkuzmin/go #gosetup
/usr/local/lib/go-1.10.4/bin/go test -c -o /private/var/folders/9g/vv960m3x0fz_yjjk5dj9wtqm0000gn/T/___TestCloneBuilderAcc_notes_in_github_com_jetbrains_infra_packer_builder_vsphere_clone github.com/jetbrains-infra/packer-builder-vsphere/clone #gosetup
/usr/local/lib/go-1.10.4/bin/go tool test2json -t /private/var/folders/9g/vv960m3x0fz_yjjk5dj9wtqm0000gn/T/___TestCloneBuilderAcc_notes_in_github_com_jetbrains_infra_packer_builder_vsphere_clone -test.v -test.run ^TestCloneBuilderAcc_notes$ #gosetup
=== RUN   TestCloneBuilderAcc_notes
2018/11/07 20:54:21 [DEBUG] Parsing template...
2018/11/07 20:54:21 [DEBUG] Initializing core...
2018/11/07 20:54:21 [DEBUG] Retrieving 'test' build
2018/11/07 20:54:21 [DEBUG] Preparing 'test' build
2018/11/07 20:54:21 [DEBUG] Running 'test' build
2018/11/07 20:54:21 Running builder: test
2018/11/07 20:54:21 ui: ==> test: Cloning VM...
--- FAIL: TestCloneBuilderAcc_notes (0.96s)
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0xf0 pc=0x17a983f]

goroutine 6 [running]:
testing.tRunner.func1(0xc42023a0f0)
	/usr/local/lib/go-1.10.4/src/testing/testing.go:742 +0x29d
panic(0x1936ae0, 0x2481fa0)
	/usr/local/lib/go-1.10.4/src/runtime/panic.go:502 +0x229
github.com/jetbrains-infra/packer-builder-vsphere/driver.(*VirtualMachine).Clone(0xc420248220, 0x1c1b240, 0xc4202a6040, 0xc42030d590, 0x0, 0x0, 0xc420270898)
	/Users/mkuzmin/go/src/github.com/jetbrains-infra/packer-builder-vsphere/driver/vm.go:224 +0x30f
github.com/jetbrains-infra/packer-builder-vsphere/clone.(*StepCloneVM).Run(0xc42029cd60, 0x1c1b240, 0xc4202a6040, 0x1c1a680, 0xc420266c90, 0x0)
	/Users/mkuzmin/go/src/github.com/jetbrains-infra/packer-builder-vsphere/clone/step_clone.go:50 +0x2c9
github.com/jetbrains-infra/packer-builder-vsphere/vendor/github.com/hashicorp/packer/helper/multistep.(*BasicRunner).Run(0xc4202a6000, 0x1c1a680, 0xc420266c90)
	/Users/mkuzmin/go/src/github.com/jetbrains-infra/packer-builder-vsphere/vendor/github.com/hashicorp/packer/helper/multistep/basic_runner.go:72 +0x22f
github.com/jetbrains-infra/packer-builder-vsphere/clone.(*Builder).Run(0xc420103320, 0x1c1bc00, 0xc42029b120, 0x1c175c0, 0xc420266c60, 0x1c1b4c0, 0xc42029b100, 0x0, 0x1c1b100, 0xc42029b0c0, ...)
	/Users/mkuzmin/go/src/github.com/jetbrains-infra/packer-builder-vsphere/clone/builder.go:80 +0x725
github.com/jetbrains-infra/packer-builder-vsphere/vendor/github.com/hashicorp/packer/packer.(*coreBuild).Run(0xc42011c0c0, 0x1c1bbc0, 0xc4200a45f0, 0x1c1b4c0, 0xc42029b100, 0x0, 0x0, 0xc42006b660, 0xc42023c030, 0x1)
	/Users/mkuzmin/go/src/github.com/jetbrains-infra/packer-builder-vsphere/vendor/github.com/hashicorp/packer/packer/build.go:230 +0x890
github.com/jetbrains-infra/packer-builder-vsphere/vendor/github.com/hashicorp/packer/helper/builder/testing.Test(0x1c1acc0, 0xc42023a0f0, 0x0, 0x1c1a640, 0xc420103320, 0xc420134d80, 0x10d, 0xc42006b830, 0x0, 0x0)
	/Users/mkuzmin/go/src/github.com/jetbrains-infra/packer-builder-vsphere/vendor/github.com/hashicorp/packer/helper/builder/testing/testing.go:154 +0x7b2
github.com/jetbrains-infra/packer-builder-vsphere/clone.TestCloneBuilderAcc_notes(0xc42023a0f0)
	/Users/mkuzmin/go/src/github.com/jetbrains-infra/packer-builder-vsphere/clone/builder_acc_test.go:582 +0x106
testing.tRunner(0xc42023a0f0, 0x1b7ee10)
	/usr/local/lib/go-1.10.4/src/testing/testing.go:777 +0xd0
created by testing.(*T).Run
	/usr/local/lib/go-1.10.4/src/testing/testing.go:824 +0x2e0

Process finished with exit code 1
```

Comment (remijouannet) [reply driver/vm.go]: woups, sorry that was pretty ugly

**Comment by jcoconnor** [comment]
👏 Tx

**Comment by tigerhawk6** [comment]
Is this working for anyone else? I've tested by building from master and from v2.1 release.

image

**Comment by mkuzmin** [comment]
not released yet, but should work in current build

**Comment by tigerhawk6** [comment]
I did a fresh build today and got the same error.

**Comment by tigerhawk6** [comment]
That version works fine. Trying to figure out why my build from master isn't. Weird. Anyway, thanks for the help.

**Comment by jcoconnor** [comment]
I've just tried build 170 and its still failing with the following error:
```
10:34:16 ==> vcenter-iso: Uploading created floppy image
10:34:16 ==> vcenter-iso: Destroying VM...
10:34:17 Build 'vcenter-iso' errored: host '' not found
10:34:17 
10:34:17 ==> Some builds didn't complete successfully and had errors:
10:34:17 --> vcenter-iso: host '' not found
```

**Comment by mkuzmin** [comment]
Check #183, this is related to clusters

**Comment by jcoconnor** [comment]
So - I tried filling in a fixed value for the ```host``` parameter and build worked fine - Notes field was filled.
Just saw your new comment - is a build available with both this and #183 addressed - happy to run thru test (although will be on PTO for about 2 during holiday period).

**Pull request merged**